Update urls

beatadelura opened the pull request:
## What is the purpose of this pull request?

Updating URLs.

---

I went through all files and checked/changed URLs. I left URLs in `test` files to avoid regression. Let me know, If you think, `test` files should be also changed.

There is one more URL to change: `http://ckeditor.com/license`, `https://cksource.com/ckeditor/license/` - but it's under discussion. 

---

Closes #1131.

AnnaTomanek commented on .github/SUPPORT.md:
```diff
@@ -17,6 +17,6 @@ For more advanced needs (e.g. CKEditor integration in your products, or developm
 For more reading visit our supporting resources:
 
-* An extensive [CKEditor 4 documentation](http://docs.ckeditor.com/) with information about CKEditor features and configuration settings.
+* An extensive [CKEditor 4 documentation](https://docs.ckeditor.com/ckeditor4/latest/) with information about CKEditor features and configuration settings.
```
The link for documentation is: https://docs.ckeditor.com/ckeditor4/docs/, https://docs.ckeditor.com/ckeditor4/latest/ is the entry page that includes SDK etc. Since we have samples separately below, I'd use the direct documentation link.

AnnaTomanek commented on .github/ISSUE_TEMPLATE.md:
```diff
@@ -3,7 +3,7 @@
 Replace this comment with your issue type: Bug / Feature request / Other, please explain.
 
-Please **do not report security issues here**, use the contact form at http://cksource.com/contact instead.
+Please **do not report security issues here**, use the contact form at https://cksource.com/contact/ instead.
```
https://ckeditor.com/contact/

AnnaTomanek commented on .github/CONTRIBUTING.md:
```diff
@@ -9,7 +9,7 @@ If you are preparing a pull request, see [Contributing Code and Providing Patche
 
 1. Create your own [plugins](http://docs.ckeditor.com/#!/guide/plugin_sdk_sample) or [skins](http://docs.ckeditor.com/#!/guide/skin_sdk_intro)
-and submit them to [CKEditor Add-Ons Repository](http://ckeditor.com/addons/plugins).
+and submit them to [CKEditor Add-Ons Repository](https://ckeditor.com/addons/plugins).
```
https://ckeditor.com/cke4/addons/plugins/

AnnaTomanek commented on core/config.js:
```diff
@@ -319,7 +319,7 @@ CKEDITOR.config = {
 	 * **Note:** The most recommended way to
 	 * [add CKEditor plugins](http://docs.ckeditor.com/#!/guide/dev_plugins) is through
-	 * [CKEditor Builder](http://ckeditor.com/builder). Read more in the
+	 * [CKEditor Builder](https://ckeditor.com/builder). Read more in the
```
https://ckeditor.com/cke4/builder

AnnaTomanek commented on core/creators/themedui.js:
```diff
@@ -309,7 +309,7 @@ CKEDITOR.replaceClass = 'ckeditor';
 	/**
 	 * Gets the element that can be used to check the editor size. This method
-	 * is mainly used by the [Editor Resize](http://ckeditor.com/addon/resize) plugin, which adds
+	 * is mainly used by the [Editor Resize](https://ckeditor.com/addon/resize) plugin, which adds
```
https://ckeditor.com/cke4/addon/resize

AnnaTomanek commented on package.json:
```diff
@@ -25,10 +25,10 @@
 	"license": "(GPL-2.0 OR LGPL-2.1 OR MPL-1.1)",
-	"bugs": "http://dev.ckeditor.com",
-	"homepage": "http://ckeditor.com",
+	"bugs": "https://dev.ckeditor.com",
```
https://github.com/ckeditor/ckeditor-dev/issues

AnnaTomanek commented on plugins/codesnippet/samples/codesnippet.html:
```diff
@@ -36,7 +36,7 @@ <h1 class="samples">
 	<div class="description">
 		<p>
-			This editor is using the <strong><a href="http://ckeditor.com/addon/codesnippet">Code Snippet</a></strong> plugin which introduces beautiful code snippets.
+			This editor is using the <strong><a href="https://ckeditor.com/addon/codesnippet">Code Snippet</a></strong> plugin which introduces beautiful code snippets.
```
https://ckeditor.com/cke4/addon/codesnippet

AnnaTomanek commented on plugins/codesnippet/samples/codesnippet.html:
```diff
@@ -154,7 +154,7 @@ <h1 class="samples">
 
 	<p>
-		The <a href="http://ckeditor.com/addon/codesnippetgeshi"><strong>Code Snippet GeSHi</strong></a> plugin is an
+		The <a href="https://ckeditor.com/addon/codesnippetgeshi"><strong>Code Snippet GeSHi</strong></a> plugin is an
```
https://ckeditor.com/cke4/addon/codesnippetgeshi

AnnaTomanek commented on plugins/colorbutton/plugin.js:
```diff
@@ -457,7 +457,7 @@ CKEDITOR.config.colorButton_backStyle = {
 /**
  * Whether the plugin should convert `background` CSS properties with color only, to a `background-color` property,
- * allowing the [Color Button](http://ckeditor.com/addon/colorbutton) plugin to edit these styles.
+ * allowing the [Color Button](https://ckeditor.com/addon/colorbutton) plugin to edit these styles.
```
https://ckeditor.com/cke4/addon/colorbutton

AnnaTomanek commented on plugins/contextmenu/plugin.js:
```diff
@@ -163,7 +163,7 @@ CKEDITOR.plugins.add( 'contextmenu', {
 
 /**
- * Whether to enable the context menu. Regardless of the setting the [Context Menu](http://ckeditor.com/addon/contextmenu)
+ * Whether to enable the context menu. Regardless of the setting the [Context Menu](https://ckeditor.com/addon/contextmenu)
```
https://ckeditor.com/cke4/addon/contextmenu

AnnaTomanek commented on plugins/language/plugin.js:
```diff
@@ -4,7 +4,7 @@
 
 /**
- * @fileOverview The [Language](http://ckeditor.com/addon/language) plugin.
+ * @fileOverview The [Language](https://ckeditor.com/addon/language) plugin.
```
https://ckeditor.com/cke4/addon/language

AnnaTomanek commented on plugins/language/plugin.js:
```diff
@@ -153,7 +153,7 @@
 /**
  * Specifies the list of languages available in the
- * [Language](http://ckeditor.com/addon/language) plugin. Each entry
+ * [Language](https://ckeditor.com/addon/language) plugin. Each entry
```
https://ckeditor.com/cke4/addon/language

AnnaTomanek commented on plugins/magicline/plugin.js:
```diff
@@ -4,7 +4,7 @@
 
 /**
- * @fileOverview The [Magic Line](http://ckeditor.com/addon/magicline) plugin that makes it easier to access some document areas that
+ * @fileOverview The [Magic Line](https://ckeditor.com/addon/magicline) plugin that makes it easier to access some document areas that
```
https://ckeditor.com/cke4/addon/magicline

AnnaTomanek commented on plugins/mathjax/plugin.js:
```diff
@@ -4,7 +4,7 @@
 
 /**
- * @fileOverview The [Mathematical Formulas](http://ckeditor.com/addon/mathjax) plugin that allows you to create and modify mathematical equations written in TeX directly in CKEditor..
+ * @fileOverview The [Mathematical Formulas](https://ckeditor.com/addon/mathjax) plugin that allows you to create and modify mathematical equations written in TeX directly in CKEditor..
```
https://ckeditor.com/cke4/addon/mathjax

AnnaTomanek commented on plugins/mathjax/plugin.js:
```diff
@@ -453,7 +453,7 @@
 /**
  * Sets the default class for `span` elements that will be
- * converted into [Mathematical Formulas](http://ckeditor.com/addon/mathjax)
+ * converted into [Mathematical Formulas](https://ckeditor.com/addon/mathjax)
```
https://ckeditor.com/cke4/addon/mathjax

AnnaTomanek commented on plugins/sharedspace/plugin.js:
```diff
@@ -138,7 +138,7 @@
  *		};
  *
- * **Note:** The [Maximize](http://ckeditor.com/addon/maximize) and [Editor Resize](http://ckeditor.com/addon/resize)
+ * **Note:** The [Maximize](https://ckeditor.com/addon/maximize) and [Editor Resize](https://ckeditor.com/addon/resize)
```
https://ckeditor.com/cke4/addon/maximize, https://ckeditor.com/cke4/addon/resize

AnnaTomanek commented on plugins/stylesheetparser/plugin.js:
```diff
@@ -4,7 +4,7 @@
 
 /**
- * @fileOverview The [Stylesheet Parser](http://ckeditor.com/addon/stylesheetparser) plugin.
+ * @fileOverview The [Stylesheet Parser](https://ckeditor.com/addon/stylesheetparser) plugin.
```
https://ckeditor.com/cke4/addon/stylesheetparser

AnnaTomanek commented on plugins/tableselection/plugin.js:
```diff
@@ -851,7 +851,7 @@
 	/**
 	 * Namespace providing a set of helper functions for working with tables, exposed by
-	 * [Table Selection](http://ckeditor.com/addon/tableselection) plugin.
+	 * [Table Selection](https://ckeditor.com/addon/tableselection) plugin.
```
https://ckeditor.com/cke4/addon/tableselection

AnnaTomanek commented on plugins/toolbar/plugin.js:
```diff
@@ -782,7 +782,7 @@ CKEDITOR.config.toolbarLocation = 'top';
  * This configuration option should not be overused. The recommended way is to use the
  * {@link CKEDITOR.config#removePlugins} setting to remove features from the editor
- * or even better, [create a custom editor build](http://ckeditor.com/builder) with
+ * or even better, [create a custom editor build](https://ckeditor.com/builder) with
```
https://ckeditor.com/cke4/builder

AnnaTomanek commented on plugins/widgetselection/plugin.js:
```diff
@@ -339,7 +339,7 @@
 
 		/**
-		 * Adds an integration for the [Select All](http://ckeditor.com/addon/selectall) plugin to the given `editor`.
+		 * Adds an integration for the [Select All](https://ckeditor.com/addon/selectall) plugin to the given `editor`.
```
https://ckeditor.com/cke4/addon/selectall

AnnaTomanek commented on plugins/wysiwygarea/plugin.js:
```diff
@@ -659,7 +659,7 @@ CKEDITOR.config.disableNativeTableHandles = true;
  * *Ctrl* key when right-clicking if {@link #browserContextMenuOnCtrl}
  * is enabled or you are simply not using the
- * [context menu](http://ckeditor.com/addon/contextmenu) plugin.
+ * [context menu](https://ckeditor.com/addon/contextmenu) plugin.
```
https://ckeditor.com/cke4/addon/contextmenu

AnnaTomanek commented on samples/index.html:
```diff
@@ -17,12 +17,12 @@
 		</ul>
 		<ul class="navigation-a-right grid-width-30">
-			<li><a href="http://ckeditor.com/blog-list">CKEditor Blog</a></li>
+			<li><a href="https://ckeditor.com/blog-list">CKEditor Blog</a></li>
```
https://ckeditor.com/blog/

AnnaTomanek commented on samples/index.html:
```diff
@@ -17,12 +17,12 @@
 	<div class="grid-container">
 		<ul class="navigation-a-left grid-width-70">
-			<li><a href="http://ckeditor.com">Project Homepage</a></li>
+			<li><a href="https://ckeditor.com">Project Homepage</a></li>
```
https://ckeditor.com/ckeditor-4/

AnnaTomanek commented on samples/toolbarconfigurator/index.html:
```diff
@@ -25,12 +25,12 @@
 	<div class="grid-container">
 		<ul class="navigation-a-left grid-width-70">
-			<li><a href="http://ckeditor.com">Project Homepage</a></li>
+			<li><a href="https://ckeditor.com">Project Homepage</a></li>
```
https://ckeditor.com/ckeditor-4/

AnnaTomanek commented on samples/toolbarconfigurator/index.html:
```diff
@@ -25,12 +25,12 @@
 		</ul>
 		<ul class="navigation-a-right grid-width-30">
-			<li><a href="http://ckeditor.com/blog-list">CKEditor Blog</a></li>
+			<li><a href="https://ckeditor.com/blog-list">CKEditor Blog</a></li>
```
https://ckeditor.com/blog/

beatadelura commented:
@AnnaTomanek Thanks for help and review! I changed requested files, PTAL. 

There is one more thing, I found file `CHANGES.md` which should be also updated. I'm working on it, so I'll let you know when PR will be ready :)

beatadelura commented:
@AnnaTomanek `CHANGES.md` updated. PTAL

AnnaTomanek commented:
Just noticed that we also have links to API docs that changed their location... For example, https://docs.ckeditor.com/#!/api/CKEDITOR.focusManager-method-focus -> https://docs.ckeditor.com/ckeditor4/docs/#!/api/CKEDITOR.focusManager-method-focus

Do we also want to change those? Should be easy enough...

beatadelura commented:
@AnnaTomanek I changed it. PTAL :)

msamsel commented:
@AnnaTomanek do you know a little bit more about this case? What kind/types of link we want to correct? Is there some kind of list (at least domains which should be corrected?).
Do you know also what's the status of `http://ckeditor.com/license` link? Is there some decision how it should looks like?

AnnaTomanek commented:
We're leaving http://ckeditor.com/license as is, a correct redirect will be in place.

msamsel commented:
@AnnaTomanek, should here remain `http` protocol or should be updated to `https`?

msamsel commented:
I've made additional url replacements. Some info:
- I tried to avoid modify `tests` folder to not break any unit test.
- Last 3 commits (c9a6933, e74f0d7, 7362194) are made manually
- Other commits has followed replacements ( links might exist with some ending url, which wasn't replace and should remain untouched ):
  * http://ckeditor.com/license -> https://ckeditor.com/license
  * http://docs.ckeditor.com/#!/ -> https://docs.ckeditor.com/ckeditor4/docs/#!/
  * http://sdk.ckeditor.com -> https://sdk.ckeditor.com
  * http://ckeditor.com/addon -> https://ckeditor.com/cke4/addon
  * http://dev.ckeditor.com -> https://dev.ckeditor.com

AnnaTomanek commented:
Looks fine, I noticed two minor http://ckeditor.com cases in samples (`enterkey` and `inlineall`).

Also plenty of http://ckeditor.com and http://cksource.com in `tests` but I guess it's intentional, so let's leave it.
